### ISSUE TITLE
word language model criterion

### ISSUE THREAD

**Issue by noowad93** [pytorch/examples#714]
https://github.com/pytorch/examples/blob/e9e76722dad4f4569651a8d67ca1d10607db58f9/word_language_model/main.py#L104

Is it okay to use cross entropy loss here?
To my understanding,`torch.nn.functional.cross_entropy` will apply log_softmax internally.
But in this codes, already applied `F.log_softmax`

**Comment by zhangguanheng66** [comment]
yeap. A PR to remove `F.log_softmax` is acceptable.

**Comment by User-LiL** [comment]
In the case of   "--model  LSMT" , does it need something like "torch.nn.Softmax"  or  "torch.nn.LogSoftmax"  operation ?

**Comment by User-LiL** [comment]
Hi, I am a begineer for NLP and pytorch.

In the case of "--model LSMT" , does it need something like "torch.nn.Softmax" or "torch.nn.LogSoftmax" operation before the computation of nn.NLLLoss( ) criterion ?

**Comment by zhangguanheng66** [comment]
CrossEntropyLoss combines `LogSoftmax` and `NLLLoss`
https://pytorch.org/docs/stable/nn.html#torch.nn.CrossEntropyLoss

**Comment by User-LiL** [comment]
In the main.py, the criterion implemented with:
          criterion = nn.NLLLoss()       #   main.py  line104
In the case of TransformerModel, there has log_softmax function:
          return F.log_softmax(output, dim=-1)     #model.py  line150
But in the case of RNNModel (--model = LSTM), I cann't find some funtion related to softmax ?
Or is it not necessary to use function related to softmax, and only torch.nn.NLLLoss( ) was enough ?

**Comment by zhangguanheng66** [comment]
Both cases need `softmax` + `NLLLoss`
